### DISCUSSION TITLE
Linker error in Visual Studio 2022

### DISCUSSION THREAD

**Discussion by jwcnmr** [alandefreitas/matplotplusplus#295] [General]
I have been trying to use Visual Studio to link one of this simple examples.
I can compile successful by putting C:\Program Files\matplotplusplus 1.1.0\include in the include path in the solution Properties window. But I am getting linker errors.
 It seems to me that I only need to link to matplot.lib which is installed in C:\Program Files\matplotplusplus 1.1.0\lib. 
But the linker is showing
Severity	Code	Description	Project	File	Line	Suppression State
Error	LNK2019	unresolved external symbol "class std::shared_ptr<class matplot::axes_type> __cdecl matplot::gca(void)" (?gca@matplot@@YA?AV?$shared_ptr@Vaxes_type@matplot@@@std@@XZ) referenced in function "auto __cdecl matplot::plot<class std::set<double,struct std::less<double>,class std::allocator<double> >,class std::vector<double,class std::allocator<double> > &,char const (&)[5]>(class std::set<double,struct std::less<double>,class std::allocator<double> >,class std::vector<double,class std::allocator<double> > &,char const (&)[5])" (??$plot@V?$set@NU?$less@N@std@@V?$allocator@N@2@@std@@AEAV?$vector@NV?$allocator@N@std@@@2@AEAY04$$CBD@matplot@@YA?A_PV?$set@NU?$less@N@std@@V?$allocator@N@2@@std@@AEAV?$vector@NV?$allocator@N@std@@@2@AEAY04$$CBD@Z)	plot3	C:\Users\James\source\repos\plot3\plot3\plot3.obj	1	

I would like to use Visual Studio as it is usually used, because I am writing a book on moving from Python to C++ and I want to make the journey as simple as possible.  I also tried including the path to nodesoup.lib under Additional Library Directories without any change.

I am sure I am missing something simple, and I would appreciate comments guiding me to getting these to compile in VS.  
Thanks
Jim Cooper

**Comment by alandefreitas** [comment]
> It seems to me that I only need to link to matplot.lib which is installed in C:\Program Files\matplotplusplus 1.1.0\lib.

The process is a little more complex than that. That's why we provide instructions for CMake. Doing it manually is a pain.

>  I want to make the journey as simple as possible

Then vcpkg is probably the best solution, even though the formula is probably not updated as often. Not only for matplot++ but anyone moving from python who needs libraries. Moving from python to C++ without CMake is only going to increase to pain for anything slightly more complex.
